src/plugins/intel_gpu/tests/test_cases/roll_gpu_test.cpp

```diff
@@ -178,22 +200,47 @@ std::vector<roll_test_params<T>> getRollFloatingPointParams() {
-                             testing::ValuesIn(func<type>()), \
+std::vector<format::type> formats4d = {format::bfyx,
+                                       format::bs_fs_yx_bsv32_fsv32,
+                                       format::bs_fs_yx_bsv32_fsv16,
+                                       format::b_fs_yx_fsv32,
+                                       format::b_fs_yx_fsv16,
+                                       format::bs_fs_yx_bsv16_fsv16};
+
+std::vector<format::type> formats5d = {format::bfzyx,
+                                       format::bs_fs_zyx_bsv32_fsv32,
+                                       format::bs_fs_zyx_bsv32_fsv16,
+                                       format::b_fs_zyx_fsv32,
+                                       format::b_fs_zyx_fsv16,
+                                       format::bs_fs_zyx_bsv16_fsv16};
+
+std::vector<format::type> formats6d = {format::bfwzyx};
+
+#define INSTANTIATE_ROLL_TEST_SUITE(type, func, formats)                                                    \
+    using roll_test_##type = roll_test<type>;                                                               \
+    TEST_P(roll_test_##type, roll_##type_##func) {                                                          \
+        test();                                                                                             \
+    }                                                                                                       \
+    INSTANTIATE_TEST_SUITE_P(roll_smoke_##type_##func,                                                      \
+                             roll_test_##type,                                                              \
+                             testing::Combine(testing::ValuesIn(func<type>()), testing::ValuesIn(formats)), \
                              roll_test_##type::PrintToStringParamName);
 
-INSTANTIATE_ROLL_TEST_SUITE(int8_t, getRollParams)
-INSTANTIATE_ROLL_TEST_SUITE(uint8_t, getRollParams)
-INSTANTIATE_ROLL_TEST_SUITE(int32_t, getRollParams)
-INSTANTIATE_ROLL_TEST_SUITE(int64_t, getRollParams)
-INSTANTIATE_ROLL_TEST_SUITE(FLOAT16, getRollFloatingPointParams)
-INSTANTIATE_ROLL_TEST_SUITE(float, getRollFloatingPointParams)
+INSTANTIATE_ROLL_TEST_SUITE(int8_t, getRollParams, formats4d)
+INSTANTIATE_ROLL_TEST_SUITE(uint8_t, getRollParams, formats4d)
+INSTANTIATE_ROLL_TEST_SUITE(int32_t, getRollParams, formats4d)
+INSTANTIATE_ROLL_TEST_SUITE(int64_t, getRollParams, formats4d)
+INSTANTIATE_ROLL_TEST_SUITE(int8_t, getRollParams5D, formats5d)
+INSTANTIATE_ROLL_TEST_SUITE(uint8_t, getRollParams5D, formats5d)
+INSTANTIATE_ROLL_TEST_SUITE(int32_t, getRollParams5D, formats5d)
+INSTANTIATE_ROLL_TEST_SUITE(int64_t, getRollParams5D, formats5d)
+INSTANTIATE_ROLL_TEST_SUITE(int8_t, getRollParams6D, formats6d)
+INSTANTIATE_ROLL_TEST_SUITE(uint8_t, getRollParams6D, formats6d)
+INSTANTIATE_ROLL_TEST_SUITE(int32_t, getRollParams6D, formats6d)
+INSTANTIATE_ROLL_TEST_SUITE(int64_t, getRollParams6D, formats6d)
```

Comment:
I think we can reduce the # of tests here too, because too many tests are TBD here. 
Also for blocked format, small shapes than block size are not helpful. 
Please pick some meaningful tests, instead of Combine with redundant tests..

Reply:
the goal to use all the blocked formats in tests is that sometimes we caught errors when we lost to add the needed pair to ocl for example. The goal to use 4d, 5D, and 6D is that we use different index extraction in the kernel for different shapes' dimensions. 
About small sizes - these tests' goal is to cover that functionality works fine, for sure blocked formats won't be used for small shapes in real life but what is sense using big shapes just to check blocked formats are adapted?

Reply:
I can remove some tests for sure but I don't know how we catch regression if somebody removes the pair type-format in ocl implementation. For 5D and 6D case we have only one input shape. What I can do is reduce the number of input shapes for blocked formats in 4d. Would it be enough?

Reply:
oh, I see duplicates through running... will fix it

Reply:
done